shared/lib_battery_dispatch_automatic_btm.cpp

```diff
@@ -640,23 +640,7 @@ void dispatch_automatic_behind_the_meter_t::plan_dispatch_for_cost(dispatch_plan
         }
     }
 
-    for (i = 0; i < _steps_per_hour && (i < sorted_grid.size()); i++)
```

Comment:
Since steps per hour was usually 1, this wasn't actually doing anything. The decrease in stored energy value associated with choosing a lower marginal cost period resulted in a substantially lower NPV.